pyiceberg/catalog/glue.py

```diff
@@ -252,6 +259,22 @@ def _construct_database_input(database_name: str, properties: Properties) -> Dat
     return database_input
 
 
+def _register_glue_catalog_id_with_glue_client(glue: GlueClient, glue_catalog_id: str) -> None:
+    """
+    Register the Glue Catalog ID (AWS Account ID) as a parameter on all Glue client methods.
+
+    It's more ergonomic to do this than to pass the CatalogId as a parameter to every client call since it's an optional
+    parameter and boto3 does not support 'None' values for missing parameters.
```

Comment:
I could (very easily) be convinced that this is indeed _not_ more ergonomic, and that it makes more sense to add the actual `CatalogId` parameter to every single invocation of a Glue client call.

That looks something like this:

```python
def _glue_kwargs(self) -> dict[str, str]:
    if self.glue_catalog_id is not None:
        return dict(CatalogId=self.glue_catalog_id)
    return dict()

...

# Add this kwarg expansion to every single client call.
self.glue.get_databases(**self._glue_kwargs())
```

It works, it just gets a little noisy.

Reply:
A third alternative would be to wrap `GlueClient` in a thin wrapper class that bakes in Glue Catalog ID and exposes just the functionality required by the caller in `GlueCatalog`. But that felt like even more work and places to forget to include `CatalogId` when adding new functions to the client wrapper.

Reply:
Thanks for providing all these possibilities. I like the current approach since it make things simple and clear